File: src/main/java/com/graqr/threshr/model/redsky/stores/RollingOperatingHours.java
package com.graqr.threshr.model.redsky.stores;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.micronaut.core.annotation.Nullable;
import io.micronaut.serde.annotation.Serdeable;

import java.util.List;

@Serdeable
public record RollingOperatingHours(
        @JsonProperty("main_hours") MainHours mainHours,
        @Nullable @JsonProperty("capability_hours")
        List<CapabilityHour> capabilityHours) {
}

File: src/main/kotlin/com/graqr/threshr/model/redsky/store/RollingOperatingHours.kt
package com.graqr.threshr.model.redsky.store

import com.fasterxml.jackson.annotation.JsonProperty
import io.micronaut.core.annotation.Nullable
import io.micronaut.serde.annotation.Serdeable

@Serdeable
data class RollingOperatingHours(
    @param:JsonProperty("main_hours")
    val mainHours: MainHours,
    @param:Nullable @param:JsonProperty("capability_hours")
    val capabilityHours: List<CapabilityHour>
) {
}
